elasticsearch-persistence/test/unit/model_store_test.rb

```diff
@@ -311,6 +311,43 @@ def valid?; false; end;
         assert subject.update( {}, { script: 'EXEC' } )
       end
 
+      should "not update an invalid model" do
+        subject.expects(:persisted?).returns(true)
+        subject.expects(:id).returns('abc123').at_least_once
+
+        @gateway
+          .expects(:update)
+          .never
+
+        subject.instance_eval do
+          def valid?; false; end;
+        end
+
+        assert ! subject.update
+        assert ! subject.persisted?
```

Comment:
Test suite of this project has never worked for me sadly so I haven't been able to outside of Travis CI